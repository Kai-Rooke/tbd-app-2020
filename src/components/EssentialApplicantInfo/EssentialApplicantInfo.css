.form-background {
    width: 100%;
    background-color: #B2DDDF;
   display: inline-block;
    padding: 20px;
  /*text-align: left;*/
    border: 1px solid black;
    margin-bottom: 10px;
}

.form-container {
    text-align: center;
}

.form-title {
    text-align: center;
}
.form-right {
    float: right;
    width: 50%;
    margin-top: 30px;
}
.form-left {
    float:left;
    width: 50%;
    padding-left: 40px;
}
@media only screen and (max-width: 768px){
    .form-left {
    width: 100%;
    float: left;
    padding-left: 0px;
    }
    .form-right {
        width: 100%;
        float: left;
    }
}

