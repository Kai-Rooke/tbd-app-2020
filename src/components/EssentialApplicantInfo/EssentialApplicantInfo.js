import React from "react";
import DropdownMenu from "../DropdownMenu/DropdownMenu";
import TextBox from "../TextBox/TextBox";
import RadioButton from "../RadioButton/RadioButton";
import "./EssentialApplicantInfo.css";
import { Container, Row, Col } from "react-bootstrap";
import UCPButton from "../Buttons/UCPButton";
<<<<<<< HEAD
import Modal from "../Modal/Modal";
=======
import {isMobile} from 'react-device-detect';
>>>>>>> main

const EssentialApplicantInfo = (props) => {
  let textboxClassName;
  if(isMobile)
  {
 
    textboxClassName = "mobiletextbox";
  }
  else{
    
    textboxClassName = "desktoptextbox";
  }

  var nationalityOptions = [
    { label: "UK National", id: "UK", value: "UK" },
    { label: "EU National", id: "EU", value: "EU" },
    { label: "Other", id: "Other", value: "Other" },
  ];

  var criminalOptions = [
    { label: "Yes", id: "crimeYes", value: "yes" },
    { label: "No", id: "crimeNo", value: "no" },
  ];

  var supportOptions = [
    { label: "Yes", id: "supportYes", value: "yes" },
    { label: "No", id: "supportNo", value: "no" },
  ];

  var time = [
      {value: 'morning', label: '09:00am - 12:00pm'},
      {value: 'monday', label: '12:00pm - 15:00pm'},
      {value: 'monday', label: '15:00pm - 17:00pm'}
    ];
  
  var days = [
      {value: 'monday', label: 'Monday'},
      {value: 'tuesday', label: 'Tuesday'},
      {value: 'wednesday', label: 'Wednesday'},
      {value: 'thursday', label: 'Thursday'},
      {value: 'friday', label: 'Friday'}
    ];

  function ReturnModalData() {
    console.log("made it here!");
    return (
    <Modal id="callback" btnText="Submit">
    <div>
        <h2>This is a modal</h2>
        <p>Example modal</p>
    </div>
</Modal>
)
  }

  function UserTypeSelect() {
    if (global.userType === "staff") {
      return (
        <Row id="buttonRow">
          <Col className="centered-buttons">
            <UCPButton
              primary="True"
              className="mediumbutton"
              buttonText="Go Back"
            />
          </Col>
          <Col className="centered-buttons">
            <div id="confirmButton">
              <UCPButton
                primary="True"
                className="mediumbutton"
                buttonText="Confirm and Go"
              />
            </div>
          </Col>
        </Row>
      );
    } else {
      return (
        <Row id="buttonRow">
          <Col className="centered-buttons">
            <div id="confirmButton">
              <UCPButton
                primary="True"
                className="mediumbutton"
                buttonText="Apply Now"
              />
            </div>
          </Col>
          <Col className="centered-buttons">
            <div id="confirmButton1" hidden>
              <Modal id="hello" btnText="Request a Call Back">
                <div>
                  <h2>This is a modal</h2>
                  <p>Example modal</p>
                  <DropdownMenu  options={days}/>
                  <DropdownMenu  options={time}/>
                  <UCPButton 
                    to="none"
                    primary="True"
                    className="smallbutton"
                    buttonText="Submit" />
                </div>
              </Modal>
            </div>
          </Col>

          <Col className="centered-buttons">
            <div id="confirmButton2">
              <UCPButton
                primary="True"
                className="mediumbutton"
                buttonText="Request Video Call Back"
              />
            </div>
          </Col>
        </Row>
      );
    }
  }
  return (
    <Container className="form-container">
      <div className="form-background">
        <h3 className="form-title">CLEARING APPLICATION</h3>
        <span class="form-left">
          <TextBox
            title="Name"
            placeholder="Enter your name"
            minLength="5"
            maxLength="60"
            id="nameinput"
            type="text"
            name="name"
            attribute="disabled"
            className=  {textboxClassName}
            
          />

          <TextBox
            title="Email"
            placeholder="Enter your email"
            minLength="5"
            maxLength="60"
            id="emailinput"
            type="email"
            name="email"
            attribute="disabled"
            className=  {textboxClassName}

          />

          <TextBox
            title="Tel No"
            placeholder="Enter phone number"
            minLength="5"
            maxLength="12"
            id="telinput"
            type="tel"
            name="telno"
            attribute="disabled"
            className=  {textboxClassName}

          />

          <TextBox
            title="D/O/B"
            placeholder="Enter date of birth"
            id="dobinput"
            type="date"
            name="dob"
            attribute="disabled"
            className=  {textboxClassName}
           
        
            
          />

          <TextBox
            title="UCAS ID"
            placeholder="Enter UCAS ID"
            id="ucasinput"
            type="text"
            name="ucas"
            attribute="disabled"
            className=  {textboxClassName}

          />
        </span>
        <span class="form-right">
          <RadioButton
            heading="Nationality"
            options={nationalityOptions}
            name="nationality"
            attribute="disabled"
          />

          <RadioButton
            heading="Convictions"
            options={criminalOptions}
            name="criminal"
            subtitle="Do you have any unspent criminal convictions?"
            attribute="disabled"
          />

          <RadioButton
            heading="Learning Support"
            options={supportOptions}
            name="support"
            subtitle="Do you have any additional learning support needs or disabilities?"
            attribute="disabled"
          />
          <h4>Preferred Course</h4>
          <DropdownMenu />
        </span>
      </div>
      <div>{UserTypeSelect()}</div>
    </Container>
  );
};

export default EssentialApplicantInfo;
