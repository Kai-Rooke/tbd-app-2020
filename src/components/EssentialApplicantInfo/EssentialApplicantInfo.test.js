import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import EssentialApplicantInfo from './EssentialApplicantInfo';

//cleans up 
afterEach(cleanup);

it("Renders Essential Applicant Info", () => {
    const { asFragment } = render(<EssentialApplicantInfo />);
    expect(asFragment()).toMatchSnapshot();
});
