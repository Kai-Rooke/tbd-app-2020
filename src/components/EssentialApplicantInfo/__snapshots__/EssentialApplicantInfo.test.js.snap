// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Essential Applicant Info 1`] = `
<DocumentFragment>
  <div
    class="form-container container"
  >
    <div
      class="form-background"
    >
      <h3
        class="form-title"
      >
        CLEARING APPLICATION
      </h3>
      <span
        class="form-left"
      >
        <div
          class="mobiletextbox"
        >
          <label
            for="nameinput"
          >
            Name
          </label>
          <input
            disabled=""
            id="nameinput"
            maxlength="60"
            minlength="5"
            name="name"
            placeholder="Enter your name"
            size="20"
            type="text"
          />
        </div>
        <div
          class="mobiletextbox"
        >
          <label
            for="emailinput"
          >
            Email
          </label>
          <input
            disabled=""
            id="emailinput"
            maxlength="60"
            minlength="5"
            name="email"
            placeholder="Enter your email"
            size="20"
            type="email"
          />
        </div>
        <div
          class="mobiletextbox"
        >
          <label
            for="telinput"
          >
            Tel No
          </label>
          <input
            disabled=""
            id="telinput"
            maxlength="12"
            minlength="5"
            name="telno"
            placeholder="Enter phone number"
            size="20"
            type="tel"
          />
        </div>
        <div
          class="mobiletextbox"
        >
          <label
            for="dobinput"
          >
            D/O/B
          </label>
          <input
            disabled=""
            id="dobinput"
            maxlength="30"
            minlength="0"
            name="dob"
            placeholder="Enter date of birth"
            size="20"
            type="date"
          />
        </div>
        <div
          class="mobiletextbox"
        >
          <label
            for="ucasinput"
          >
            UCAS ID
          </label>
          <input
            disabled=""
            id="ucasinput"
            maxlength="30"
            minlength="0"
            name="ucas"
            placeholder="Enter UCAS ID"
            size="20"
            type="text"
          />
        </div>
      </span>
      <span
        class="form-right"
      >
        <div
          class="radio-buttons"
        >
          <h4>
            Nationality
          </h4>
          <p />
          <div
            class="radio-group"
          >
            <input
              disabled=""
              id="UK"
              name="nationality"
              type="radio"
              value="UK"
            />
            <label
              for="UK"
            >
              UK National
            </label>
          </div>
          <div
            class="radio-group"
          >
            <input
              disabled=""
              id="EU"
              name="nationality"
              type="radio"
              value="EU"
            />
            <label
              for="EU"
            >
              EU National
            </label>
          </div>
          <div
            class="radio-group"
          >
            <input
              disabled=""
              id="Other"
              name="nationality"
              type="radio"
              value="Other"
            />
            <label
              for="Other"
            >
              Other
            </label>
          </div>
        </div>
        <div
          class="radio-buttons"
        >
          <h4>
            Convictions
          </h4>
          <p>
            Do you have any unspent criminal convictions?
          </p>
          <div
            class="radio-group"
          >
            <input
              disabled=""
              id="crimeYes"
              name="criminal"
              type="radio"
              value="yes"
            />
            <label
              for="crimeYes"
            >
              Yes
            </label>
          </div>
          <div
            class="radio-group"
          >
            <input
              disabled=""
              id="crimeNo"
              name="criminal"
              type="radio"
              value="no"
            />
            <label
              for="crimeNo"
            >
              No
            </label>
          </div>
        </div>
        <div
          class="radio-buttons"
        >
          <h4>
            Learning Support
          </h4>
          <p>
            Do you have any additional learning support needs or disabilities?
          </p>
          <div
            class="radio-group"
          >
            <input
              disabled=""
              id="supportYes"
              name="support"
              type="radio"
              value="yes"
            />
            <label
              for="supportYes"
            >
              Yes
            </label>
          </div>
          <div
            class="radio-group"
          >
            <input
              disabled=""
              id="supportNo"
              name="support"
              type="radio"
              value="no"
            />
            <label
              for="supportNo"
            >
              No
            </label>
          </div>
        </div>
        <h4>
          Preferred Course
        </h4>
        <div
          class="Dropdown css-2b097c-container"
        >
          <div
            class=" css-yk16xz-control"
          >
            <div
              class=" css-g1d714-ValueContainer"
            >
              <div
                class=" css-1wa3eu0-placeholder"
              >
                Select...
              </div>
              <div
                class="css-b8ldur-Input"
              >
                <div
                  class=""
                  style="display: inline-block;"
                >
                  <input
                    aria-autocomplete="list"
                    autocapitalize="none"
                    autocomplete="off"
                    autocorrect="off"
                    id="react-select-2-input"
                    spellcheck="false"
                    style="box-sizing: content-box; width: 2px; border: 0px; font-size: inherit; opacity: 1; outline: 0; padding: 0px;"
                    tabindex="0"
                    type="text"
                    value=""
                  />
                  <div
                    style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-size: inherit; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                  />
                </div>
              </div>
            </div>
            <div
              class=" css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class=" css-1okebmr-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class=" css-tlfecz-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-6q0nyr-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </span>
    </div>
    <div>
      <div
        class="row"
        id="buttonRow"
      >
        <div
          class="centered-buttons col"
        >
          <div
            id="confirmButton"
          >
            <a
              href="/emptyLink"
            >
              <button
                class="mediumbutton"
                data-testid="ucpButton"
                primary="True"
                type="button"
              >
                Apply Now​
              </button>
            </a>
          </div>
        </div>
        <div
          class="centered-buttons col"
        >
          <div
            id="confirmButton1"
          >
            <a
              href="/emptyLink"
            >
              <button
                class="mediumbutton"
                data-testid="ucpButton"
                primary="True"
                type="button"
              >
                Request Call Back​
              </button>
            </a>
          </div>
        </div>
        <div
          class="centered-buttons col"
        >
          <div
            id="confirmButton2"
          >
            <a
              href="/emptyLink"
            >
              <button
                class="mediumbutton"
                data-testid="ucpButton"
                primary="True"
                type="button"
              >
                Request Video Call Back​
              </button>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
